packages/rpc-core/src/rpc-subscriptions/slots-updates-notifications.ts

```diff
@@ -1,9 +1,18 @@
 import { Slot, U64UnsafeBeyond2Pow53Minus1 } from '../rpc-methods/common';
 
 type SlotsUpdatesNotificationsApiNotification = Readonly<{
-    // Only present for 'createdBank' and notifications
+    // Only present for 'dead' notifications
+    err?: string;
+    // Only present for 'createdBank' notifications
     parent?: Slot;
     slot: Slot;
+    // Only present for 'frozen' notifications
+    stats?: Readonly<{
+        maxTransactionsPerEntry: U64UnsafeBeyond2Pow53Minus1;
+        numFailedTransactions: U64UnsafeBeyond2Pow53Minus1;
+        numSuccessfulTransactions: U64UnsafeBeyond2Pow53Minus1;
+        numTransactionEntries: U64UnsafeBeyond2Pow53Minus1;
+    }>;
     timestamp: U64UnsafeBeyond2Pow53Minus1;
     type: 'completed' | 'createdBank' | 'dead' | 'firstShredReceived' | 'frozen' | 'optimisticConfirmation' | 'root';
 }>;
```

Comment:
Is it worth splitting these out into the notification types a bit?

Something like this:

```ts
type SlotsUpdatesNotificationsApiNotificationBase = Readonly<{
  slot: Slot;
  timestamp: U64UnsafeBeyond2Pow53Minus1;
  type: 'completed' | 'firstShredReceived' | 'optimisticConfirmation' | 'root';
}>;

type SlotsUpdatesNotificationsApiNotificationDead = SlotsUpdatesNotificationsApiNotificationBase & Readonly<{
  err?: string;
  type: 'dead' // educated guess!
}>;

type SlotsUpdatesNotificationsApiNotificationCreatedBank = SlotsUpdatesNotificationsApiNotificationBase & Readonly<{
  parent: Slot; // or still `parent?`
  type: 'createdBank'
}<;

type SlotsUpdatesNotificationsApiNotificationFrozen = SlotsUpdatesNotificationsApiNotificationBase & Readonly<{
      stats: Readonly<{ // maybe still `stats?`
        maxTransactionsPerEntry: U64UnsafeBeyond2Pow53Minus1;
        numFailedTransactions: U64UnsafeBeyond2Pow53Minus1;
        numSuccessfulTransactions: U64UnsafeBeyond2Pow53Minus1;
        numTransactionEntries: U64UnsafeBeyond2Pow53Minus1;
    }>;
    type: 'frozen'
}>;

type SlotsUpdatesNotificationsApiNotification = SlotsUpdatesNotificationsApiNotificationBase | SlotsUpdatesNotificationsApiNotificationDead | SlotsUpdatesNotificationsApiNotificationCreatedBank | SlotsUpdatesNotificationsApiNotificationFrozen
```

Then you'd be able to branch on `type` and only get the relevant fields when you receive one of these notifications

Reply:
Yeah, I wasn't totally sure if this was necessary, but looking at it laid out I think it's good to do. Added!

Reply:
the Rust implementation doesn't seem to use any options, so I'm taking these fields as guaranteed when the notification type is set